services/processor/native/repository/_Elections/processing_rewards.go

```diff
@@ -145,38 +144,48 @@ func getExcellenceProgramGuardians() []byte {
 	return state.ReadBytes(_formatExcellenceProgramGuardians())
 }
 
-func _setExcellenceProgramGuardians(guardians map[[20]byte]uint64) {
+func _setExcellenceProgramGuardians(guardians guardianArray) {
 	guardiansForSave := make([]byte, 0, len(guardians)*20)
-	for guardianAddr := range guardians {
-		guardiansForSave = append(guardiansForSave, guardianAddr[:]...)
+	for _, guardian := range guardians {
+		guardiansForSave = append(guardiansForSave, guardian.address[:]...)
 	}
 	state.WriteBytes(_formatExcellenceProgramGuardians(), guardiansForSave)
 }
 
 /***
  * Rewards: Sort top guardians using sort.Interface
  */
-func _getTopGuardians(guardiansAccumulatedStake map[[20]byte]uint64) (topGuardiansStake map[[20]byte]uint64, totalVotes uint64) {
+func _getTopGuardians(guardiansAccumulatedStake map[[20]byte]uint64) (topGuardiansStake guardianArray, totalVotes uint64) {
 	totalVotes = uint64(0)
-	topGuardiansStake = make(map[[20]byte]uint64)
 
 	guardianList := make(guardianArray, 0, len(guardiansAccumulatedStake))
 	for guardian, vote := range guardiansAccumulatedStake {
 		guardianList = append(guardianList, &guardianVote{guardian, vote})
 	}
 	sort.Sort(guardianList)
 
-	for i := 0; i < ELECTION_GUARDIAN_EXCELLENCE_MAX_NUMBER; i++ {
-		fmt.Printf("elections %10d rewards: top guardian %x, has %d votes\n", _getCurrentElectionBlockNumber(), guardianList[i].guardian, guardianList[i].vote)
+	i := 0
```

Comment:
if you must define outside `var i int` is better because you set it explicitly to 0 later. but i would also define it as a different variable name and do `topRewardQualifyingGuardIdx = i` inside the loop or something..